src/components/media-player-track.js

```diff
@@ -29,24 +29,28 @@ function MediaPlayerTrack({
   onPlayStateChange
 }) {
   const playerRef = useRef();
-  const [lastProgress, setLastProgress] = useState(0);
+  const [sourceDuration, setSourceDuration] = useState(0);
   const [lastSeekTimestamp, setLastSeekTimestamp] = useState(0);
-  const [durationInMilliseconds, setDurationInMilliseconds] = useState(0);
+  const [lastProgressTimecode, setLastProgressTimecode] = useState(0);
   const [currentPlayState, setCurrentPlayState] = useState(MEDIA_PLAY_STATE.initializing);
 
   const changePlayState = newPlayState => {
     setCurrentPlayState(newPlayState);
     onPlayStateChange?.(newPlayState);
   };
 
-  const changeProgress = newProgress => {
-    setLastProgress(newProgress);
-    onProgress?.(newProgress - (startTimecode || 0));
+  const changeProgress = newSourceTimecode => {
+    const trackStartTimecode = playbackRange[0] * sourceDuration;
+    setLastProgressTimecode(newSourceTimecode);
+    onProgress?.(newSourceTimecode - trackStartTimecode);
   };
 
-  const seekToStartIfNecessary = duration => {
-    if (duration && ((startTimecode && lastProgress < startTimecode) || (stopTimecode && lastProgress >= stopTimecode))) {
-      playerRef.current.seekTo((startTimecode || 0) / duration);
+  const seekToStartIfNecessary = newSourceDuration => {
+    const newTrackStartTimecode = playbackRange[0] * newSourceDuration;
+    const newTrackStopTimecode = playbackRange[1] * newSourceDuration;
+
+    if (newSourceDuration && ((lastProgressTimecode < newTrackStartTimecode) || (lastProgressTimecode >= newTrackStopTimecode))) {
+      playerRef.current.seekTo(newTrackStartTimecode / newSourceDuration);
```

Comment:
Here I think it makes sense to early return if `!newSourceDuration`, cause the calculation of the 2 values and the `if` evaluation are not needed otherwise. Would make the code more clear too.

Reply:
Good idea, we do that in most other places anyway, so it aligns. Done.